tasks/internal_reconcile_report_phantom/internal_reconcile_report_phantom.py

```diff
@@ -0,0 +1,204 @@
+import json
+from http import HTTPStatus
+from typing import Any, Dict, List, Union
+
+import fastjsonschema as fastjsonschema
+from cumulus_logger import CumulusLogger
+from fastjsonschema import JsonSchemaException
+from orca_shared.database import shared_db
+from orca_shared.database.shared_db import retry_operational_error
+from sqlalchemy import text
+from sqlalchemy.future import Engine
+
+OUTPUT_JOB_ID_KEY = "jobId"
+OUTPUT_ANOTHER_PAGE_KEY = "anotherPage"
+OUTPUT_PHANTOMS_KEY = "phantoms"
+PHANTOMS_COLLECTION_ID_KEY = "collectionId"
+PHANTOMS_GRANULE_ID_KEY = "granuleId"
+PHANTOMS_FILENAME_KEY = "filename"
+PHANTOMS_KEY_PATH_KEY = "keyPath"
+PHANTOMS_ORCA_ETAG_KEY = "orcaEtag"
+PHANTOMS_ORCA_LAST_UPDATE_KEY = "orcaLastUpdate"
+PHANTOMS_ORCA_SIZE_KEY = "orcaSize"
+
+LOGGER = CumulusLogger()
+
+PAGE_SIZE = 100
+
+
+def task(
+    job_id: int,
+    page_index: int,
+    db_connect_info: Dict[str, str],
+) -> Dict[str, Any]:
+    """
+    Args:
+        job_id: The unique ID of job/report.
+        page_index: The 0-based index of the results page to return.
+        db_connect_info: See requests_db.py's get_configuration for further details.
+    """
+    engine = shared_db.get_user_connection(db_connect_info)
+    granules = query_db(
+        engine,
+        job_id,
+        page_index,
+    )
+
+    return {
+        OUTPUT_JOB_ID_KEY: job_id,
+        OUTPUT_ANOTHER_PAGE_KEY: len(granules) > PAGE_SIZE,
+        OUTPUT_PHANTOMS_KEY: granules[0:PAGE_SIZE],  # we get one extra for anotherPage calculation.
+    }
+
+
+@retry_operational_error()
+def query_db(
+    engine: Engine,
+    job_id: str,
+    page_index: int,
+) -> List[Dict[str, Any]]:
+    """
+    Gets phantoms for the given job/page, up to PAGE_SIZE + 1 results.
+
+    Args:
+        engine: The sqlalchemy engine to use for contacting the database.
+        job_id: The unique ID of job/report.
+        page_index: The 0-based index of the results page to return.
+
+    Returns:
+        A list containing dicts matching the format of "phantoms" in output.json.
+    """
+    LOGGER.info(f"Retrieving page '{page_index}' of reports for job '{job_id}'")
+    with engine.begin() as connection:
+        sql_results = connection.execute(
+            get_phantoms_sql(),
+            [
+                {
+                    "job_id": job_id,
+                    "page_index": page_index,
+                    "page_size": PAGE_SIZE,
+                }
+            ],
+        )
+
+        phantoms = []
+        for sql_result in sql_results:
```

Comment:
I haven't tested this but any issues if no data returned? In other words no phantoms exist for the given job_id? From a REST perspective, would we return a 404 not found error?

Reply:
Would seem odd to me to have users wrap each call to our API in try/catch just for the case when no error was found. I'd be open to returning 404 if the job for the job_id is incomplete/non-existent, but that would add an extra sql query.

Reply:
I don't believe you would need an extra query, you should be able to just check if the result set returned for phantoms is empty.

Reply:
Nope. A valid report can have an empty list. Would seem odd to me to have users wrap each call to our API in try/catch just for the case when no error was found.

Reply:
Leave as is for now as we may look at changes down the road if we switch to GraphQL or some other solution.